pkg/network/kafka/batch_manager.go

```diff
@@ -0,0 +1,122 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package kafka
+
+import (
+	"errors"
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+)
+
+var errLostBatch = errors.New("kafka batch lost (not consumed fast enough)")
+
+const maxLookupsPerCPU = 2
```

Comment:
what does that mean?

Reply:
Copied it from the HTTP data path, it's used in `GetPendingTransactions` for bound the number of lookups in the batch map per CPU

Reply:
I think it's fine to leave as it is considering that this is a fork and this code will be replaced by https://github.com/DataDog/datadog-agent/pull/14712 but the motivation for having multiple CPU lookups was to improve the amount of events we return when "syncing" with kernel-space, because we'll potentially scrape more eBPF map entries that haven't been flushed via the perf buffer. And @DanielLavie is right in saying that this is to limit the number of lookups, because otherwise we could enter in a sort of loop where kernel space is "churning" batch entries so fast that userspace keeps trying to keep up indefinitely by issuing more eBPF map lookups.